packages/ingest/src/consumerRateLimited.ts

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-await-in-loop */
+import { receiveSQSMessages, SQSMessage } from '@cumulus/aws-client/SQS';
+import * as sqs from '@cumulus/aws-client/SQS';
+import { ExecutionAlreadyExists } from '@cumulus/aws-client/StepFunctions';
+import Logger from '@cumulus/logger';
+import { sleep } from '@cumulus/common';
+
+export type MessageConsumerFunction = (queueUrl: string, message: SQSMessage) => Promise<void>;
+
+const log = new Logger({ sender: '@cumulus/ingest/consumer' });
+export interface ConsumerConstructorParams {
+  queueUrls: string[],
+  timeRemainingFunc: () => number,
+  visibilityTimeout: number,
+  deleteProcessedMessage?: boolean,
+  rateLimitPerSecond: number;
+}
+
+export class ConsumerRateLimited {
+  private readonly deleteProcessedMessage: boolean;
+  private readonly queueUrls: string[];
+  private readonly timeRemainingFunc: () => number;
+  private readonly visibilityTimeout: number;
+  private readonly rateLimitPerSecond: number;
+  private readonly timeBuffer: number;
+  private readonly messageLimitPerFetch: number;
+  private readonly waitTime: number;
+
+  constructor({
+    queueUrls,
```

Comment:
Nit: consider commenting all of the class params in the interface,  using optional parameters for the params that are being set to static ingegers and doing something like 

e.g. 
```
  /**
   * URLs of the SQS queues to poll.
   */
  queueUrls: string[];
```

and so on in the interface.       Totally get the original consumer didn't have this.

Reply:
Oh, actually @Jkovarik can you clarify what you mean by `using optional parameters for the params that are being set to static integers`?  I was thinking you were saying;
```
private readonly timeBuffer?: number;
private readonly messageLimitPerFetch?: number;
private readonly waitTime?: number;
```
But I think I'm misinterpreting since they're always set to values so aren't optional.